mirroring/handler.go

```diff
@@ -0,0 +1,230 @@
+package mirroring
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/go-openapi/strfmt"
+	trilliantypes "github.com/google/trillian/types"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/spf13/viper"
+)
+
+// consider adding a rekorClient field so that the same client struct is used for all operations, potentially saving some time
+type LogHandler struct {
+	metadata        TreeMetadata
+	client          *client.Rekor
+	newLeavesBuffer []Artifact
+}
+
+func LoadFromRemote(serverURL string) (h LogHandler, err error) {
```

Comment:
Not entirely sure what this API funcion gives me. Is this part of a CLI entrypoint to load a remote MHT? is that happening automagically on `GetLogInfo` ? or on `h.client` ?

Reply:
I thought that the two main construction scenarios for the log-handler would be to either:
-Fill in the public key, root hash, tree size and other metadata from a previously stored state
or
-Given a provided URL, to fetch the metadata from a Rekor log.

For the second case (LoadFromRemote), GetLogInfo is called to fetch the metadata and store it in the log-handler, and then a Rekor client is initiated so that for almost all other operations where a Rekor API call is needed, the client is not instantiated again and again, allowing for a gain in performance.